File: 056_Power_Digit_Sum/Solution.go
package main

import (
	"fmt"
	"math/big"
	"strconv"
)

func main() {
	var num int
	fmt.Scan(&num)

	maxSum := 0

	for i := 1; i<num; i++ {
		for j := 1; j<num; j++ {
			if maxSum < getPowerSumOfDigits(i,j) {
				maxSum = getPowerSumOfDigits(i,j)
			}
		}
	}

	fmt.Println(maxSum)

}

func getPowerSumOfDigits(a, b int) int {
	pow := new(big.Int).Exp(big.NewInt(int64(a)), big.NewInt(int64(b)), nil)
	sum := 0
	for _, char := range pow.String() {
		digit, err := strconv.Atoi(string(char))
		if err != nil {
			fmt.Println("Error converting character to integer:", err)
			continue
		}
		sum += digit
	}

	return sum
}

File: 056_Power_Digit_Sum/Solution.py


def get_pow_val_sum(a, b):
    out = a ** b
    _sum = 0
    for var in str(out):
        _sum += int(var)
    return _sum

num = int(input().strip())



max_sum = 0


for a in range(1,num):
    for b in range(1, num):
        if max_sum < get_pow_val_sum(a,b):
            max_sum = get_pow_val_sum(a,b)
        
print(max_sum)